tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_opcodes.py

```diff
@@ -375,159 +406,1216 @@ def test_bal_extcodesize_and_oog(
-    bob = pre.fund_eoa()
+
+    target = (
+        pre.empty_account()
+        if target_is_empty
+        else pre.deploy_contract(code=Op.STOP)
+    )
+
+    ret_size = 32 if memory_expansion else 0
+
+    call_code = Op.CALL(
+        gas=0, address=target, value=value, ret_size=ret_size, ret_offset=0
+    )
+    caller = pre.deploy_contract(code=call_code, balance=value)
+
+    access_list = (
+        [AccessList(address=target, storage_keys=[])]
+        if target_is_warm
+        else None
+    )
+
+    intrinsic_cost = fork.transaction_intrinsic_cost_calculator()(
+        access_list=access_list
+    )
+
+    bytecode_cost = gas_costs.G_VERY_LOW * 7
+
+    access_cost = (
+        gas_costs.G_WARM_ACCOUNT_ACCESS
+        if target_is_warm
+        else gas_costs.G_COLD_ACCOUNT_ACCESS
+    )
+    transfer_cost = gas_costs.G_CALL_VALUE if value > 0 else 0
+    memory_cost = fork.memory_expansion_gas_calculator()(new_bytes=ret_size)
+
+    # Create cost: only if value > 0 AND target is empty
+    create_cost = (
+        gas_costs.G_NEW_ACCOUNT if (value > 0 and target_is_empty) else 0
+    )
+
+    # static gas (before state access): access + transfer + memory
+    static_gas_cost = access_cost + transfer_cost + memory_cost
+    # second check includes create_cost
+    second_check_cost = static_gas_cost + create_cost
+
+    if oog_boundary == OutOfGasBoundary.OOG_BEFORE_TARGET_ACCESS:
+        gas_limit = intrinsic_cost + bytecode_cost + static_gas_cost - 1
+    else:  # SUCCESS
+        gas_limit = intrinsic_cost + bytecode_cost + second_check_cost
+
+    tx = Transaction(
+        sender=alice,
+        to=caller,
+        gas_limit=gas_limit,
+        access_list=access_list,
+    )
+
+    # BAL expectations
+    account_expectations: Dict[Address, BalAccountExpectation | None]
+    if oog_boundary == OutOfGasBoundary.OOG_BEFORE_TARGET_ACCESS:
+        # Target NOT in BAL - we OOG before state access
+        account_expectations = {
+            caller: BalAccountExpectation.empty(),
+            target: None,
+        }
+    elif value > 0:
+        account_expectations = {
+            caller: BalAccountExpectation(
+                balance_changes=[BalBalanceChange(tx_index=1, post_balance=0)]
+            ),
+            target: BalAccountExpectation(
+                balance_changes=[
+                    BalBalanceChange(tx_index=1, post_balance=value)
+                ]
+            ),
+        }
+    else:
+        account_expectations = {
+            caller: BalAccountExpectation.empty(),
+            target: BalAccountExpectation.empty(),
+        }
+
+    value_transferred = value > 0 and oog_boundary == OutOfGasBoundary.SUCCESS
+
+    post_state: Dict[Address, Account | None] = {alice: Account(nonce=1)}
+
+    if value_transferred:
+        post_state[target] = Account(balance=value)
+        post_state[caller] = Account(balance=0)
+    else:
+        post_state[caller] = Account(balance=value)
```

Comment:
Nit: in this case `value = 0`, so we can put `Account()`? (Note: it still has code)